tortoisebeacon/handlers.go

```diff
@@ -148,16 +147,51 @@ func (tb *TortoiseBeacon) handleProposalMessage(ctx context.Context, m ProposalM
 	return nil
 }
 
-func (tb *TortoiseBeacon) isPotentiallyValidProposalMessage(atxTimestamp time.Time, nextEpochStart time.Time, receivedTimestamp time.Time) bool {
+func (tb *TortoiseBeacon) verifyProposalMessage(m ProposalMessage, currentEpoch types.EpochID) (done bool, err error) {
```

Comment:
why do we need `done` boolean here? seems like in every other option we return an error

Reply:
It means that message processing is done. If `done == true`, we need to return from the outer function. We may need to return from the outer function even if there's no error, e.g. if a proposal does not pass an eligibility threshold.

Reply:
if we are done `handleProposalMessage` will exit with error message anyways so calling function will receive error in any case so `handleProposalMessage` can basically check if error was received and return it in